modules/solid_mechanics/src/interfacekernels/AnalysisStepUOInterface.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "AnalysisStepUOInterface.h"
+#include "AnalysisStepUserObject.h"
+
+void
+AnalysisStepUOInterface::getAnalysisStepUserObject(const FEProblemBase & fe_problem,
+                                                   const AnalysisStepUserObject *& step_user_object,
+                                                   const std::string & name)
+{
```

Comment:
seems like this whole interface could be replaced with a `getUniqueUserObject<>` templated routine in the UO interface

Reply:
@GiudGiud This is probably outside the scope of this PR, mainly because I don't know how to do this